packages/datadog_flutter_plugin/android/src/main/kotlin/com/datadoghq/flutter/DatadogRumPlugin.kt

```diff
@@ -275,6 +286,217 @@ class DatadogRumPlugin(
             )
         }
     }
+
+    @Suppress("TooGenericExceptionCaught")
+    internal fun <T> callEventMapper(
+        mapperName: String,
+        event: T,
+        encodedEvent: Map<String, Any?>,
+        completion: (Map<String, Any?>?, T) -> T?
+    ): T? {
+        var modifiedJson: Map<String, Any?>? = encodedEvent
+        val latch = CountDownLatch(1)
+
+        val handler = Handler(Looper.getMainLooper())
+        handler.post {
+            try {
+                channel.invokeMethod(
+                    mapperName,
+                    mapOf(
+                        "event" to encodedEvent
+                    ),
+                    object : Result {
+                        override fun success(result: Any?) {
+                            modifiedJson = (result as? Map<String, Any?>)
+                            latch.countDown()
+                        }
+
+                        override fun error(
+                            errorCode: String,
+                            errorMessage: String?,
+                            errorDetails: Any?
+                        ) {
+                            latch.countDown()
+                        }
+
+                        override fun notImplemented() {
+                            Datadog._internal._telemetry.error(
+                                "$mapperName returned notImplemented."
+                            )
+                            latch.countDown()
+                        }
+                    }
+                )
+            } catch (e: Exception) {
+                Datadog._internal._telemetry.error("Attempting call $mapperName failed.", e)
+                latch.countDown()
+            }
+        }
+
+        try {
+            // Stalls until the method channel finishes
+            if (!latch.await(1, TimeUnit.SECONDS)) {
+                Datadog._internal._telemetry.debug("$mapperName timed out")
+                return event
+            }
+
+            if (modifiedJson?.containsKey("_dd.mapper_error") == true) {
+                return event
+            }
+
+            return completion(modifiedJson, event)
+        } catch (e: Exception) {
+            Datadog._internal._telemetry.error(
+                "Attempt to deserialize mapped log event failed, or latch await was interrupted." +
+                    " Returning unmodified event.",
+                e
```

Comment:
Here it might be interesting to distinguish both cases to have a more accurate message. I'm not sure what the best practice is in dart, but I guess using two `catch` blocks, one for the "deserialzation failed" and one for the "latch await interrupted". 
Especially since the deserialization indicates a SDK issue and could be sent to telemetry, whereas latch await would mean a usage issue.